test/e2e/tests/book-appointment.spec.ts

```diff
@@ -0,0 +1,91 @@
+  const today: Date = new Date();
+  const curMonth: string = today.toLocaleString('default', { month: 'short' });
+  const curYear: string = String(today.getFullYear());
+  await expect(bookingPage.calendarHeader).toHaveText(`${curMonth} ${curYear}`);
+  // confirm button is disabled by default until a slot is selected
+  await expect(bookingPage.confirmBtn).toBeDisabled();
+}
+
+test.beforeEach(async ({ page }) => {
+  bookingPage = new BookingPage(page);
+  dashboardPage = new DashboardPage(page);
+});
+
+// verify we are able to book an appointment using existing user's share link
+test.describe('book an appointment', {
+  tag: '@prod-sanity'
+}, () => {
+  test('able to access booking page via short link', async ({ page }) => {
+    await bookingPage.gotoBookingPageShortUrl();
+    await verifyBookingPageLoaded();
+  });
+
+  test('able to access booking page via long link', async ({ page }) => {
+    await bookingPage.gotoBookingPageLongUrl();
+    await verifyBookingPageLoaded();
+  });
+
+  test('able to request a booking', async ({ page }) => {
+    // in order to ensure we find an available slot we can click on, first switch to week view URL
+    await bookingPage.gotoBookingPageWeekView();
+    await expect(bookingPage.titleText).toBeVisible({ timeout: 30_000 });
+
+    // now select an available booking time slot  
+    const selectedSlot: string|null = await bookingPage.selectAvailableBookingSlot(PROD_DISPLAY_NAME);
+    console.log(`selected appointment time slot: ${selectedSlot}`);
+
+    // now we have an availble booking time slot selected, click confirm button
+    await bookingPage.confirmBtn.click();
+
+    // now fill out the book selection dialog with booking requester's info and book it
+    await bookingPage.finishBooking(APPT_BOOKING_REQUESTER_NAME, APPT_BOOKING_REQUESTER_EMAIL);
+
+    // 'boooking request sent' text appears twice, once in the pop-up and once in underlying page
+    await expect(bookingPage.requestSentTitleText.first()).toBeVisible({ timeout: 60_000 });
+    await expect(bookingPage.requestSentTitleText.nth(1)).toBeVisible();
+
+    // booking request sent dialog availability text contains correct user name
+    // this text also appears twice, once in the pop-up and once in underlying page
+    await expect(bookingPage.requestSentAvailabilityText.first()).toBeVisible();
+    await expect(bookingPage.requestSentAvailabilityText.nth(1)).toBeVisible();
+    const expectedText: string = `${PROD_DISPLAY_NAME}'s Availability`;
+    expect(bookingPage.requestSentAvailabilityText.first()).toContainText(expectedText);
+    expect(bookingPage.requestSentAvailabilityText.nth(1)).toContainText(expectedText);
+
+    // booking request sent dialog should display the correct time slot that was requested
+    // our requested time slot is stored in this format, as example: 'event-2025-01-14 14:30'
+    // the dialog reports the slot in this format, as example: 'Tuesday, January 14, 2025 02:30 PM'
```

Comment:
do you know what code generates the time displayed in the dialog? can that logic be turned into a util and used instead of writing the getTimeFromSlotString custom function?

Reply:
Hmmm interesting idea, thanks! Another thing that might be easier or more consistent with test automation would be to add a `data-testid` attribute to the available time slots and it could be formatted like

`data-testid='Tuesday, January 14, 2025 02:30 PM'` then when the test selects an available time slot it could grab the data-testid attribute value for the selected time slot; and it would already be formatted for what we need.

Reply:
Looks like the code that generates the event time in the request sent dialog is actually generated right in the frontend itself:

https://github.com/thunderbird/appointment/blob/main/frontend/src/components/BookingModal.vue#L54

`const time = computed(() => dj(props.event.start).format(`dddd, MMMM D, YYYY ${timeFormat()}`));`

So I don't think we can go that route.

Reply:
@malini Would you mind approving this as it is? I don't see any easy way to get rid of the custom `getTimeFromSLotString` function unfortunately.